lightning/src/ln/offers_tests.rs

```diff
@@ -438,6 +440,63 @@ fn prefers_more_connected_nodes_in_blinded_paths() {
 	}
 }
 
+/// Tests the dummy hop behavior of Offers based on the message router used:
+/// - Compact paths (`DefaultMessageRouter`) should not include dummy hops.
+/// - Node ID paths (`NodeIdMessageRouter`) may include 0 to [`MAX_DUMMY_HOPS_COUNT`] dummy hops.
+#[test]
+fn check_dummy_hop_pattern_in_offer() {
+	let chanmon_cfgs = create_chanmon_cfgs(2);
+	let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+	let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+	create_announced_chan_between_nodes_with_value(&nodes, 0, 1, 10_000_000, 1_000_000_000);
+
+	let alice = &nodes[0];
+	let deterministic_entropy = Arc::new(FixedEntropy);
+
+	// Case 1: DefaultMessageRouter → uses compact blinded paths (via SCIDs)
+	// Expected: No dummy hops; each path contains only the recipient.
+	let default_router = DefaultMessageRouter::new(alice.network_graph, deterministic_entropy.clone());
```

Comment:
No need for an `Arc`. Just use a normal reference.

```suggestion

	// Case 1: DefaultMessageRouter → uses compact blinded paths (via SCIDs)
	// Expected: No dummy hops; each path contains only the recipient.
	let default_router = DefaultMessageRouter::new(alice.network_graph, &FixedEntropy);
```

Reply:
Done! [pr3726.15](https://github.com/shaavan/rust-lightning/commits/pr3726.15)
Thanks for the pointer!